### ISSUE TITLE
Deprecate CustomScrollbar

### ISSUE THREAD

**Issue by kaydelaney** [grafana/grafana#80429]
For years we've used the `CustomScrollbar` component to manage scrollbar behavior in Grafana across browsers.
We did this, I assume, because vendor support for scrollbar customization was spotty across browsers, with Firefox implementing the standard [scrollbar-width](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width) and [scrollbar-color](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color) properties, and WebKit browsers using [non-standard CSS properties](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar).
Later this month however, Chrome and other WebKit-based browsers will be adding support for the standard scrollbar CSS properties that Firefox has long supported, so I thought it was a good opportunity to reevaluate if we really need a javascript-based scrollbar solution.
I think not having to rely on javascript when possible is good practice in general, and to be honest there have been times when `CustomScrollbar` has interfered with my ability to reason about what's going on with the CSS of various components, so removing that layer of abstraction could be a help in that regard.

Anyway, first, can we achieve the scrollbar styles we want just relying on CSS? Let's compare.
For the below comparisons, the left scroll area uses the existing `CustomScrollbar` component, and the right relies purely on CSS:

### Firefox (macOS Sonoma 14.1, Scroll bars always showing):

https://github.com/grafana/grafana/assets/45561153/b18ec1ea-39ff-4cb2-b912-b481d6dd8f2e

To my eyes, the scrollbars are almost identical, down to the fade-in/out effect on hover. There _is_ some padding around the CSS-only scroll thumb, but I think this actually looks better... Plus, with the CSS-only solution, we get the stretchy/bouncy effect on the scroll thumb as with native scrollbars.

### Chrome (macOS Sonoma 14.1, Scroll bars always showing):

https://github.com/grafana/grafana/assets/45561153/4cae2f68-b856-48ea-84ac-e87d39aaac1b

Again, quite similar. The main difference here is that with WebKit's scrollbar customization, it's difficult to achieve the fade-in/out effect on the scroll (there are some tricks you can use to achieve it, but not without any downsides). I think this is fairly minor though, and since Chrome and others are on track to adopt the standard CSS properties, in the longterm this won't be an issue.

Something else to note is that various platforms allow for OS-level management of scrollbar behavior. On macOS for example, we have the option of always showing scrollbars, or only showing them when scrolling:
<img width="328" alt="image" src="https://github.com/grafana/grafana/assets/45561153/a1862c6f-5b90-474c-be7d-c9fdbff648ed">

Let's see how this affects Firefox and Chrome:

### Firefox (macOS Sonoma 14.1, Scroll bars only when scrolling):

https://github.com/grafana/grafana/assets/45561153/efb41c76-77dd-4ec3-87dd-201ae26bc158

### Chrome (macOS Sonoma 14.1, Scroll bars only when scrolling):

https://github.com/grafana/grafana/assets/45561153/4b9a42ac-af6e-4dde-a288-31de4b704221

Interesting! In both Firefox and Chrome, when the OS scrollbar behavior is set to only show them when scrolling, `CustomScrollbar` doesn't actually respect the styles we've specified, whereas the CSS-only solution _does_, which is another win for that approach in my book.

To sum up, I think ditching the `react-custom-scrollbars-2`-powered `CustomScrollbar` component in favor of simply using CSS scrollbar customization would be a good move with very few if any downsides.
Let me know your thoughts!

**Comment by torkelo** [comment]
@kaydelaney great investigation! Just so you know CustomScrollbar should revert to native scrollbars on OSX when overlay scrollbars is enabled. 

Beyond styling (IE not having white scrollbars in dark theme) the main reason for CustomScrollbars is get overlay scrollbars on windows and linux that does not support native overlay scrollbars. Overlay scrollbars has some great advantages, takes less space (important for small panels/panes), and require no width change (something that can cause pages to jump / reflow when moving between pages/views with and without scollbarr). 

Not sure what the current state is for native support for overflow scrollbars on windows/linux

**Comment by joshhunt** [comment]
(just putting thoughts from slack into the thread)

Extremely supportive of this - this is something we've been discussing quietly for a while, so it's great to have this down 'officially'. This is inline with frontend ops's soft-goal of doing less special stuff, and being more boring 🎉 

I believe where possible, Grafana should respect the platform it's running on and the user's settings. I believe it is more valuable to the user that Grafana acts like everything else on their computer, rather than Grafana forcing a specific style of scroll bar.

You're right that modern css, like `scrollbar-width` and [`color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme), give us a nice middleground of controlling aspects of the scrollbar appearance, whilst still remaining native to the OS.

For page-level scroll bars (when scrolling the whole page), I think this is an easy win and should be relatively straight forward.

For control-level scroll (such as scrolling Selects or dropdowns), or even in places like Panel Edit, it might require more thought, but I think it's something we should start to move towards.

**Comment by joshhunt** [comment]
Also related https://github.com/grafana/grafana/issues/13437
